### DISCUSSION TITLE
Setting CPU clock to 2.0 GHz on Lichee Pi 4A-16G

### DISCUSSION THREAD

**Discussion by cmd0s** [revyos/revyos#85] [General]
Hi everyone,

I need help setting the **CPU** clock frequency to **2.0 GHz** on my **Lichee Pi 4A-16G**. I’ve tried modifying the .dtb/.dts files, but so far, it hasn’t worked. I also attempted to search online for a solution, but unfortunately, I couldn’t find anything that helped.

Here’s some system info:

OS: Linux **revyos**-lpi4a 5.10.113-th1520 #**2024.07.20**.13.28+d8f77de53

If anyone has experience with this board or knows how to properly adjust the CPU clock settings, I’d greatly appreciate your guidance.

Thank you in advance!

**Comment by FIFCC** [comment]
Theoretically, modifying the device tree is feasible, but it is not 100% successful. In some cases, the system may crash due to chip problems. Perhaps you can show how you modified it to see if it is correct?

**Comment by FIFCC** [comment]
> Hi everyone,
> 
> I need help setting the **CPU** clock frequency to **2.0 GHz** on my **Lichee Pi 4A-16G**. I’ve tried modifying the .dtb/.dts files, but so far, it hasn’t worked. I also attempted to search online for a solution, but unfortunately, I couldn’t find anything that helped.
> 
> Here’s some system info:
> 
> OS: Linux **revyos**-lpi4a 5.10.113-th1520 #**2024.07.20**.13.28+d8f77de53
> 
> If anyone has experience with this board or knows how to properly adjust the CPU clock settings, I’d greatly appreciate your guidance.
> 
> Thank you in advance!

Have you read this article? It may help you.
https://github.com/sipeed/sipeed_wiki/blob/main/docs/hardware/zh/lichee/th1520/lpi4a/6_peripheral.md#cpu%E8%B6%85%E9%A2%91

**Reply by cmd0s** [reply]
Thank you for the link. I will go through the process and let you know if it worked.

**Reply by FIFCC** [reply]
Hi, how's the testing going? By the way, we've released a new image; feel free to give it a try!

**Reply by cmd0s** [reply]
Hi,

Thank you for the link. I installed the newest image today, but unfortunately, running `sudo ls -lh /boot` gives me the following:

```sh
debian@revyos-lpi4a:~$ sudo ls -lh /boot
total 70M
-rw-r--r-- 1 root root 263K Dec 27 11:31 config-6.6.66-th1520
drwxr-xr-x 3 root root 4.0K Dec 29 11:27 dtbs
drwxr-xr-x 2 root root 4.0K Dec 29 11:29 extlinux
-rw-r--r-- 1 root root 265K Sep 20 17:02 fw_dynamic.bin
-rw-r--r-- 1 root root  28M Jan 12 17:58 initrd.img-6.6.66-th1520
-rwxr-xr-x 1 root root  52K Sep 20 16:28 light_aon_fpga.bin
-rwxr-xr-x 1 root root 5.1M Sep 20 16:28 light_c906_audio.bin
drwx------ 2 root root  16K Dec 29 11:24 lost+found
-rwxr-xr-x 1 root root 1.7K Sep 20 16:27 str.bin
-rw-r--r-- 1 root root 3.5M Dec 27 11:31 System.map-6.6.66-th1520
-rwxr-xr-x 1 root root  34M Dec 27 11:31 vmlinux-6.6.66-th1520
```

I don’t have `light-lpi4a_2Ghz.dtb`. Instead, I see this file:
```
/boot/dtbs/linux-image-6.6.66-th1520/thead/th1520-lichee-pi-4a-16g.dtb
```

However, I still don’t know how to set my CPU to run at 2 GHz. Could you please provide more details?

**Reply by cmd0s** [reply]
After an extensive search, I discovered the file `light-lpi4a_2Ghz.dtb` in one of the versions of the boot partition image. I decompiled it using the command:  `dtc -I dtb -O dts -o file.dts file.dtb`

Both files are included in the attached `.zip` package.  
[2GHz dtb and dts.zip](https://github.com/user-attachments/files/18390627/2GHz.dtb.and.dts.zip)

I need guidance on how to properly merge them and which file to replace.
